### PR TITLE
4 Gherkin UI login logout

### PR DESCRIPTION
**Pull request by abrahimi93**
Creating a **Gherking test Case**:

-  Login
- Logout

### CODE REVIEW

File: src/pages/ui/login.page.ts
Comment (devpro) [inline src/pages/ui/login.page.ts:1]: Can you revert this change?
It will fail in the tests as the environment is up-to-date and which introduced a breaking change.
FYI I have fixed this in a pull request that is going to be merged today (it will be managed from configuration).

Comment (abrahimi93) [reply src/pages/ui/login.page.ts:1]: Okey, I will take a look

File: acceptance/ui/step-definitions/user-auth-ui.step-definition.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@binding([SessionContext])
+export class UserAuthStepDefinition extends BaseStepDefinition {
+  public constructor(protected sessionContext: SessionContext) {
+    super(sessionContext);
+  }
+
+  @given(/I am on the UI login page/, 'UI', 100000)
+  public async givenLoginPageIsOpened(): Promise<void> {
+    const loginPage = new LoginPage(this.sessionContext.page ?? (() => { throw new Error('page is null'); })());
+    await loginPage.goto(process.env.HOBBYFARM_UI_URL as string, '/login?returnUrl=%2Fapp%2Fhome');
```
Comment (devpro) [inline acceptance/ui/step-definitions/user-auth-ui.step-definition.ts:16]: For the login, you should reproduce the `UiFlow` to be able to manage the changes between version.

```ts
let homePage = await UiFlow.login(this.sessionContext.page ?? (() => { throw new Error('page is null'); })());
```

File: acceptance/ui/step-definitions/user-auth-ui.step-definition.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  @then(/I should be redirected to the UI home page/, 'UI')
+  public thenHomePageIsDisplayed(): void {
+    expect(this.sessionContext.current).toBeInstanceOf(HomePage);
+  }
+
+  @when(/the UI user is logged in the UI/)
+  public async givenUserIsLoggedIn(): Promise<void> {
+    const loginPage = new LoginPage(this.sessionContext.page ?? (() => { throw new Error('page is null'); })());
+    await loginPage.goto(process.env.HOBBYFARM_UI_URL as string, '/app/home');
```
Comment (devpro) [inline acceptance/ui/step-definitions/user-auth-ui.step-definition.ts:34]: `UiFlow` should be used to manage changes in versions

File: acceptance/ui/step-definitions/user-auth-ui.step-definition.ts (hunk 1 above)
Comment (devpro) [reply acceptance/ui/step-definitions/user-auth-ui.step-definition.ts:16]: I'm making the code for Admin UI scenario on #18
